app/views/samples/_form_js.haml

```diff
@@ -77,3 +90,14 @@
       $('.choose-picture').removeClass('on');
     }
   }, false);
+
+  function toggleTooltipAction() {
```

Comment:
This name is not entirely correct, it's not a toggle. I'd go with `updateAddAssaysControls`.